src/server/gateways/postgres/plugin.ts

```diff
@@ -36,6 +36,7 @@ export default class PostgresGateway extends Gateway {
     } catch (err) {
       // To-do: We should be handling this according to the design.
       await client.query(TRANSACTIONS.ROLLBACK);
+      this.logger.log("error", "Id lookup failed");
```

Comment:
**Comment**: We should be logging the actual error object here, instead of a simplified message. Without the original stack trace, we will never be able to retrace the error, and thus logging negative cases would be useless.

Reply:
Changed arguments for this.logger.log to "error", "Error:", err. This will now display the error's original message, and log the full error stack.